handlers/mesh_test.go

```diff
@@ -31,48 +33,31 @@ func checkStatus(t *testing.T, expectedCode int, resp *http.Response) {
 	}
 }
 
-func TestGetMesh(t *testing.T) {
+func TestGetMeshGraph(t *testing.T) {
 	require := require.New(t)
 
 	conf := config.NewConfig()
 	kubernetes.SetConfig(t, *conf)
 
-	k8s := kubetest.NewFakeK8sClient(
-		&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: conf.IstioNamespace}},
-	)
-	business.SetupBusinessLayer(t, k8s, *conf)
+	clients := map[string]kubernetes.ClientInterface{
+		conf.KubernetesConfig.ClusterName: kubetest.NewFakeK8sClient(
+			&corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: conf.IstioNamespace}},
+		),
+	}
+	cf := kubetest.NewFakeClientFactory(conf, clients)
+	cache := cache.NewTestingCacheWithFactory(t, cf, *conf)
+	grafana := grafana.NewService(conf, clients[conf.KubernetesConfig.ClusterName])
+	discovery := istio.NewDiscovery(clients, cache, conf)
+	business.SetupBusinessLayer(t, clients[conf.KubernetesConfig.ClusterName], *conf)
+	business.WithKialiCache(cache)
+	business.WithDiscovery(discovery)
 
 	authInfo := &api.AuthInfo{Token: "test"}
-	server := httptest.NewServer(handlers.WithAuthInfo(authInfo, handlers.GetMesh))
+	handler := handlers.MeshGraph(conf, cf, cache, grafana, discovery)
+	server := httptest.NewServer(handlers.WithAuthInfo(authInfo, handler))
 	t.Cleanup(server.Close)
 
-	resp, err := http.Get(server.URL + "/api/mesh")
+	resp, err := http.Get(server.URL + "/api/mesh/graph")
 	require.NoError(err)
 	checkStatus(t, http.StatusOK, resp)
 }
-
-type forbiddenFake struct{ kubernetes.ClientInterface }
-
-func (f *forbiddenFake) GetNamespace(namespace string) (*corev1.Namespace, error) {
-	return nil, fmt.Errorf("forbidden")
-}
-
-func TestGetMeshWithoutAccess(t *testing.T) {
```

Comment:
So this is actually something we might need to address separately. I removed the `/api/mesh` endpoint because it wasn't being used anywhere and that's what this test was for. But the `/api/mesh/graph` endpoint does currently do any access checks. It seems like we should check that the user has access to at least one of the istio system namespaces? It's not entirely clear to me what the behavior should be since you can have multiple controlplanes across different namespaces so which one should we check?

Reply:
Yeah, that's interesting and I'm not 100% sure what we should allow/block.  I don't want to be overly aggressive in blocking users but checking read access to the control plane ns is probably a good idea.